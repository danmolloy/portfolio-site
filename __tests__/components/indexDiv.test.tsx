import IndexDiv, {IndexSectionProps} from "@/components/indexDiv";
import { render, screen } from "@testing-library/react";

const mockProps: IndexSectionProps = {
  children: <div data-testid="mock-child" />,
  title: "mock title",
  name: "mock name"
}

describe("<IndexDiv />", () => {
  beforeEach(() => {
    render(<IndexDiv {...mockProps} />)
  })
  it("index-div is in the document with name attr", () => {
    const indexDiv = screen.getByTestId(`index-div`)
    expect(indexDiv).toBeInTheDocument()
    //expect(indexDiv).toHaveAttribute("name", mockProps.name)
  })
  it("title is in the document", () => {
    const title = screen.getByText(mockProps.title)
    expect(title).toBeInTheDocument()
  })
  it("children is in the document", () => {
    const children = screen.getByTestId("mock-child")
    expect(children).toBeInTheDocument()
  })
  it("matches snapshot", () => {
    const indexDiv = screen.getByTestId(`index-div`)
    expect(indexDiv).toMatchSnapshot()
  })
})