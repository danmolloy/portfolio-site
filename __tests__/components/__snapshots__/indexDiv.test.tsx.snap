// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IndexDiv /> matches snapshot 1`] = `
<div
  class=" p-12 text-sm flex flex-col items-center w-full "
  data-testid="index-div"
>
  <div
    class="flex flex-row w-full"
  >
    <h1
      class="font-display my-4"
    >
      mock title
    </h1>
  </div>
  <div
    data-testid="mock-child"
  />
</div>
`;
